pkg/git/url_test.go

```diff
@@ -37,3 +38,15 @@ func TestEquivalent(t *testing.T) {
 		assert.Equal(t, u.equal, r.Equivalent(u.equivalent))
 	}
 }
+
+func TestHasValidHostname(t *testing.T) {
+	for _, url := range []string{
+		"git@github.com:fluxcd/flux",
+		"https://user@example.com:5050/repo.git",
```

Comment:
There are more valid options out there, at least: `ssh://git@example.com/org/repo.git`, `ssh://git@example.com:22/org/repo.git`, `ssh://example.com/org/repo.git`, `https://example.com/org/repo.git`.

Your logic falls short for those options:

```console
=== RUN   TestHasValidHostname
--- FAIL: TestHasValidHostname (0.00s)
    url_test.go:53: git repo URL ssh://git@example.com/org/repo.git contains invalid hostname example.com/org/repo.git
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1
```

```diff
diff --git a/pkg/git/url_test.go b/pkg/git/url_test.go
index 32d854e8..52f36ca1 100644
--- a/pkg/git/url_test.go
+++ b/pkg/git/url_test.go
@@ -43,6 +43,10 @@ func TestHasValidHostname(t *testing.T) {
        for _, url := range []string{
                "git@github.com:fluxcd/flux",
                "https://user@example.com:5050/repo.git",
+               "ssh://git@example.com/org/repo.git",
+               "ssh://git@example.com:22/org/repo.git",
+               "ssh://example.com/org/repo.git",
+               "https://example.com/org/repo.git",
        } {
                u := Remote{url}
                if hostname, isValid := u.HasValidHostname(); !isValid {
```

Reply:
hi @hiddeco, I will update this module.